tests/functional/monitoring/prometheus/alerts/test_alert_mds_cache_high_usage.py

```diff
@@ -0,0 +1,234 @@
+        )
+
+
+def active_mds_alert_values(threading_lock):
+    active_mds = cluster.get_active_mds_info()["mds_daemon"]
+    sr_mds = cluster.get_mds_standby_replay_info()["mds_daemon"]
+    cache_alert = constants.ALERT_MDSCACHEUSAGEHIGH
+    message = f"High MDS cache usage for the daemon mds.{active_mds}."
+    description = (
+        f"MDS cache usage for the daemon mds.{active_mds} has exceeded above 95% of the requested value."
+        f" Increase the memory request for mds.{active_mds} pod."
+    )
+    runbook = (
+        "https://github.com/openshift/runbooks/blob/master/alerts/"
+        "openshift-container-storage-operator/CephMdsCacheUsageHigh.md"
+    )
+    state = "firing"
+    severity = "critical"
+
+    api = prometheus.PrometheusAPI(threading_lock=threading_lock)
+    alerts_response = api.get("alerts", payload={"silenced": False, "inhibited": False})
+    prometheus_alerts = alerts_response.json()["data"]["alerts"]
+
+    prometheus.verify_mds_alerts(
+        alert_name=cache_alert,
+        msg=message,
+        description=description,
+        runbook=runbook,
+        state=state,
+        severity=severity,
+        alerts=prometheus_alerts,
+        active_mds=active_mds,
+        standby_mds=sr_mds,
+    )
+    log.info("Alert verified successfully")
+    return True
+
+
+@tier2
+@blue_squad
+class TestMdsMemoryAlerts:
+    alert_timer = 900  # sleep time to generate the alert 15 minutes
+    scale_timer = 100  # sleep time to wait after running scale down
+
+    @pytest.mark.polarion_id("OCS-5570")
+    def test_alert_triggered(
+        self, run_metadata_io_with_cephfs, threading_lock, alert_timer
+    ):
+        # This function verifies the mds cache alert triggered or not
+        log.info(
+            "Metadata IO started in the background. Script will sleep for 15 minutes before validating the MDS alert"
+        )
+        time.sleep(alert_timer)
+        log.info("Validating the alert now")
+        assert active_mds_alert_values(threading_lock)
+
+    @pytest.mark.polarion_id("OCS-5571")
+    def test_mds_cache_alert_with_active_node_drain(
+        self, run_metadata_io_with_cephfs, threading_lock, alert_timer
+    ):
+        # This function verifies the mds cache alert with active mds running node drain
+        log.info(
+            "Metadata IO started in the background. Lets wait for 15 minutes before validating the MDS alert"
+        )
+        time.sleep(alert_timer)
+        log.info("Validating the alert now")
+        assert active_mds_alert_values(threading_lock)
+
+        node_name = cluster.get_active_mds_info()["node_name"]
+        # Unschedule active mds running node.
+        unschedule_nodes([node_name])
+        log.info(f"node {node_name} unscheduled successfully")
+        # Drain node operation
+        drain_nodes([node_name])
+        log.info(f"node {node_name} drained successfully")
+        # Make the node schedule-able
+        schedule_nodes([node_name])
```

Comment:
how do you make sure that, the app pod running node is not picked here?

Reply:
We need to bring down a node where Active MDS pod is running. So the same  node name given for node drain, the same node name is giving for schedule_nodes as well.